### PR TITLE
Add CVE-2024-6633 and Fix some issues with previous detection scripts.

### PR DESCRIPTION
**Pull request by nmasdoufi-ol**
Detection for CVE-2024-6633 with some fixed for previous CVEs
- Failed unit test in [cve_2019_12989.py](https://github.com/Ostorlab/agent_asteroid/pull/106/files#diff-5871846f44b48646dca6fbb686ac300f2e3c296d0d8045b761ae5d7a4ef92043)
- Fixing the parameter that the method `connect_to_licence_server` takes in [cve_2024_38077.py](https://github.com/Ostorlab/agent_asteroid/pull/106/files#diff-26323c1686b6a0248d66bcb68b5929cd549221b3daac5d59a8ce0fa3de573932)

### CODE REVIEW

File: tests/exploits/cve_2024_6633_test.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        ), f"Expected False, but got {result}. The function should return False when the target is not vulnerable."
+
+
+def testDetectVulnerability_whenPortClosed_reportFalse() -> None:
+    """Test _detect_vulnerability when the port is closed."""
+    with mock.patch("agent.exploits.cve_2024_6633._is_port_open", return_value=False):
+        result = cve_2024_6633._detect_vulnerability("192.168.1.1")
+        assert (
+            result is False
+        ), f"Expected False, but got {result}. The function should return False when the port is closed."
```
Comment (PiranhaSa) [inline tests/exploits/cve_2024_6633_test.py:171]:
```suggestion
        result = cve_2024_6633._detect_vulnerability("192.168.1.1")
        
        assert (
            result is False
        ), f"Expected False, but got {result}. The function should return False when the port is closed."
```

File: tests/exploits/cve_2024_6633_test.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+def testDetectVulnerability_whenPortOpenButNotVulnerable_reportFalse() -> None:
+    """Test _detect_vulnerability when the port is open but the target is not vulnerable."""
+    with mock.patch(
+        "agent.exploits.cve_2024_6633._is_port_open", return_value=True
+    ), mock.patch(
+        "agent.exploits.cve_2024_6633._attempt_db_connection", return_value=False
+    ):
+        result = cve_2024_6633._detect_vulnerability("192.168.1.1")
+        assert (
+            result is False
```
Comment (PiranhaSa) [inline tests/exploits/cve_2024_6633_test.py:161]:
```suggestion
        result = cve_2024_6633._detect_vulnerability("192.168.1.1")
        
        assert (
            result is False
```

File: agent/exploits/cve_2024_6633.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    "Fortra FileCatalyst Workflow versions less than 5.1.7 are vulnerable to a critical issue where "
+    "default credentials for the internal HSQLDB are exposed. Attackers can exploit this by accessing "
+    "the database remotely using these credentials via TCP ports 4406 and 9001."
+)
+
+DEFAULT_PORTS = [4406, 9001]
+DEFAULT_USER = "sa"
+DEFAULT_PASSWORD = ""
+
+logging.basicConfig(level=logging.DEBUG)
```
Comment (deadly-panda) [inline agent/exploits/cve_2024_6633.py:25]: not needed..

**Review by ybadaoui-ostorlab — APPROVED** [review]

File: agent/exploits/cve_2024_38077.py [hunk 4]
```diff
@@ -147,7 +147,7 @@ def accept(self, target: definitions.Target) -> bool:
         Returns:
             bool: True if the target is likely vulnerable, False otherwise.
         """
-        return _connect_to_license_server(target.origin) is not None
+        return _connect_to_license_server(target.host) is not None
```
Comment (3asm) [inline agent/exploits/cve_2024_38077.py:150]: Why the change?

File: agent/exploits/cve_2019_12989.py [hunk 5]
```diff
@@ -77,7 +76,7 @@ def check(self, target: definitions.Target) -> list[definitions.Vulnerability]:
                 return [vulnerability]
             else:
                 return []
-        except json.JSONDecodeError:
+        except requests.exceptions.JSONDecodeError:
```
Comment (3asm) [inline agent/exploits/cve_2019_12989.py:79]: Why the change?

File: requirement.txt [hunk 6]
```diff
@@ -8,4 +8,6 @@ packaging
 pygments
 semantic_version
 impacket
-pysnmp==4.4.6
\ No newline at end of file
+pysnmp==4.4.6
+jaydebeapi
```
Comment (3asm) [inline requirement.txt:12]: Why add these?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: agent/exploits/cve_2024_38077.py (hunk 4 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_38077.py:150]: the `_connect_to_license_server` only needs the ip not the whole schema.

File: tests/test-requirement.txt [hunk 7]
```diff
@@ -6,4 +6,6 @@ typing-extensions
 requests_mock
 types-requests
 pytest_mock
-ruff
\ No newline at end of file
+ruff
+jaydebeapi
+jpype1
```
[the remark this answers is not in the document]
Comment (nmasdoufi-ol) [reply tests/test-requirement.txt]: just forgot and  put them there. they're removed now.

File: agent/exploits/cve_2019_12989.py (hunk 5 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2019_12989.py:79]: the unit test was failing.

File: requirement.txt (hunk 6 above)
Comment (nmasdoufi-ol) [reply requirement.txt:12]: `pysnmp` was added for another CVE. As for `jaydebeapi`, it's needed to connect to the DB.

```python
def _attempt_db_connection(db_host: str, port: int) -> bool:
    """Attempt to connect to the database using the specified port."""
    try:
        conn = jaydebeapi.connect(
            "org.hsqldb.jdbcDriver",
            f"jdbc:hsqldb:hsql://{db_host}:{port}/hsqldb",
            [DEFAULT_USER, DEFAULT_PASSWORD],
            "../Resources/hsqldb.jar",
        )
        conn.close()
        return True
```

File: agent/exploits/cve_2024_6633.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+    try:
+        conn = jaydebeapi.connect(
+            "org.hsqldb.jdbcDriver",
+            f"jdbc:hsqldb:hsql://{db_host}:{port}/hsqldb",
+            [DEFAULT_USER, DEFAULT_PASSWORD],
+            "../Resources/hsqldb.jar",
+        )
+        conn.close()
+        return True
+    except Exception as e:
```
Comment (ohachimOs) [inline agent/exploits/cve_2024_6633.py:84]: It's better we have specific exception we try an catch

Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_6633.py:84]: Indeed, I couldn't find some particular errors but I'll dig deeper.

File: agent/exploits/cve_2019_12989.py (hunk 5 above)
Comment (3asm) [reply agent/exploits/cve_2019_12989.py:79]: That's a red flag, how do I know this is the actual exception raised and not some wrong artifact of the unit test?

File: agent/exploits/cve_2024_6633.py (hunk 8 above)
Comment (3asm) [reply agent/exploits/cve_2024_6633.py:84]: +1

File: agent/exploits/cve_2019_12989.py (hunk 5 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2019_12989.py:79]: It's what was shown to me when I ran the unit tests locally, I'll leave it as It was. If it raises again, then we solve it.

File: agent/exploits/cve_2024_6633.py (hunk 8 above)
Comment (nmasdoufi-ol) [reply agent/exploits/cve_2024_6633.py]: That was fixed. Please rereview again.

**Review by 3asm — APPROVED** [review]

**Pull request merged**